django_dramatiq/apps.py

```diff
@@ -35,7 +35,21 @@ class DjangoDramatiqConfig(AppConfig):
     name = "django_dramatiq"
     verbose_name = "Django Dramatiq"
 
+    def import_models(self):
+        """
+        Initialize Dramatiq configuration right after Django has loaded its models.
+
+        Django calls `ready()` on all apps only after it has loaded all models.
+        Due to the fact that tasks often imported from models, we need to make sure
+        that Dramatiq is fully configured before that happens.
+        """
+        super().import_models()
+        self.ready()
+
     def ready(self):
+        if getattr(self, "_is_ready", False):
+            super().ready()
```

Comment:
Can you explain this a little bit? Is this needed because now that we're calling `ready` from `import_models`, it's also likely to be called during the regular django app lifecycle? If so, wouldn't it be better to avoid running the code at all the second time it's called? I.e.:

```python
def ready(self):
  if getattr(self, "_is_ready", False):
    return
  super().ready()
  ...
  self._is_ready = True
```

Reply:
Yes, the idea is to initialize this code during `import_models`, and then not to rerun it again. Your note is correct, in this case we don't want anything to be called second time, including `super().ready()`. I missed it as `AppConfig.ready` is not doing anything, but good point!